flambe/experiment/experiment.py

```diff
@@ -274,8 +274,12 @@ def run(self, force: bool = False, verbose: bool = False, debug: bool = False, *
 
         # By default use all CPUs if no GPU is present
         devices = self.devices if self.devices else None
-        if devices is None and utils.local_has_gpu():
-            devices = {"cpu": 4, "gpu": 1}
+        if devices is None:
+            cluster_devices = ray.cluster_resources()
+            if 'GPU' in cluster_devices or 'gpu' in cluster_devices:
+                devices = {"cpu": 4, "gpu": 1}
+            else:
+                devices = {"cpu": 1}
 
```

Comment:
I would extract these 4 lines into a utility method and add unit tests, as its critical to maintain the correct functionality.

Reply:
:+1: on adding tests!

Reply:
Since this code is completely going away in the next version, can I get a pass here? This fixes a big bug, and this code is very minimal